### ISSUE TITLE
support raw utf8 unicode string

### ISSUE THREAD

**Issue by ilovenwd** [mitmproxy/mitmproxy#283]
if a json response contains utf8 encoded unicode like "金", 
it will convert to "\u94dc" in JSON view
and shows "..." in raw view
This behavior make inspect much harder.
It should just show "金" in both JSON view and raw view

**Comment by ushuz** [comment]
@mhils Sorry to bother you from an ancient issue. 

This issue was closed last September, but non-ascii contents are still unicode-encoded in JSON view with mitmproxy 0.17.

![image](https://cloud.githubusercontent.com/assets/1430856/15844547/ceac8e58-2c9d-11e6-936b-e5636638e786.png)

Is there a switch or something to turn `ensure_ascii` off? Or it's currently impossible?

**Comment by dwfreed** [comment]
Currently impossible; could be implemented, but I'd highly recommend against it, because of all the nasty things you can do with Unicode (see zalgo, for example).  If you trust that what you're viewing is not going to do anything nasty, you can edit the raw body, and (assuming your editor is vim) then run
```
:%!python -c 'import json, sys; print(json.dumps(json.load(sys.stdin), sort_keys=True, indent=4, ensure_ascii=False).encode("utf-8"))'
```

**Comment by ushuz** [comment]
@dwfreed Thanks for your advice. I tried `Raw` mode after reading your comment, and I found that non-ascii content can be displayed in `Raw` mode, and in `JavaScript` mode. 

![image](https://cloud.githubusercontent.com/assets/1430856/15845930/7a637f2a-2caa-11e6-98d2-80f6ba2d4761.png)

So it seems that it's already implemented, but not in `JSON` mode, why? Is it a bug? Or there are some further considerations?
